pkg/autoscaler/config/config.go

```diff
@@ -131,16 +135,24 @@ func validate(lc *autoscalerconfig.Config) (*autoscalerconfig.Config, error) {
 		return nil, fmt.Errorf("target-burst-capacity must be either non-negative or -1 (for unlimited), was: %f", lc.TargetBurstCapacity)
 	}
 
-	if lc.ContainerConcurrencyTargetFraction <= 0 || lc.ContainerConcurrencyTargetFraction > 1 {
+	if lc.ContainerConcurrencyTargetFraction < 0 || lc.ContainerConcurrencyTargetFraction > 1 {
 		return nil, fmt.Errorf("container-concurrency-target-percentage = %f is outside of valid range of (0, 100]", lc.ContainerConcurrencyTargetFraction)
 	}
 
-	if x := lc.ContainerConcurrencyTargetFraction * lc.ContainerConcurrencyTargetDefault; x < autoscaling.TargetMin {
+	if lc.TargetUtilization <= 0 || lc.TargetUtilization > 1 {
+		return nil, fmt.Errorf("target-utilization = %f is outside of valid range of (0, 100]", lc.TargetUtilization)
+	}
```

Comment:
what happens if both `ContainerConcurrencyTargetFraction` and `TargetUtilization` are set by the user in the configmap, should we add a check and a message here?

Reply:
I have shared more detail about this pr.
> Both TargetUtilization and ContainerConcurrencyTargetFraction are defined. ContainerConcurrencyTargetFraction will take effect in concurrency mode and TargetUtilization will take effect in rps mode. 

Agree with add a message here if both ContainerConcurrencyTargetFraction and TargetUtilization are set. But I don't know how to properly get a logger instance here.

Reply:
Sorry, without any further context, I really would get confused by a error message that I would get for a `lc.TargetUtilization == 50` : 

```
target-utilization = 50 is outside of valid range of (0, 100]
```

because that message is just not true.

Reply:
@rhuss How did you get that error message. Like the parse of `TargetFraction`, `TargetUtilization` is divided by `100.0` if it is bigger than `1.0`.

Reply:
Validated that `50` does not cause an error

Reply:
I'm just inferring from the error message above:

```golang
if lc.TargetUtilization <= 0 || lc.TargetUtilization > 1 {
    return nil, fmt.Errorf("target-utilization = %f is outside of valid range of (0, 100]", lc.TargetUtilization)
}
```

IMO the message should be either `outside of valid range of (0,1.0]` or you should multiply `lc.TargetUtilization` with 100 before setting it into the message. Otherwise you get messages like

```
target-utilization = 1.1 is outside of valid range of (0, 100]
```

if `lc.TargetUtilization == 1.1` (but 1.1 *is* in the range of (0,100])

It's just a matter of wording of the error message (not about the logic itself)